### PR TITLE
doc: v12.1.0 major changes for release

### PR DESCRIPTION
**Pull request by theanalyst**
Adding a section for major changes in 12.1.0 release, sorted by sections
so that we can have a consolidated announce when announcing the release

Adding this as a separate PR from the commit log which makes it hard to read as the diff will be too large for rendering the github UI

### CODE REVIEW

**Comment by dillaman** [comment]
Some RBD features of note for luminous:

- The default 'rbd' pool is no longer created automatically during cluster creation. Additionally, the name of the default pool used by the rbd CLI when no pool is specified can be overridden via a new ```rbd default pool = <pool name>``` configuration option.
- Initial support for deferred image deletion via new ```rbd trash``` CLI commands. Images, even ones actively in-use by clones, can be moved to the trash and deleted at a later time.
- New pool-level ```rbd mirror pool promote``` and ```rbd mirror pool demote``` commands to batch promote/demote all mirrored images within a pool.
- Mirroring now optionally supports a configurable replication delay via the ```rbd mirroring replay delay = <seconds>``` configuration option.  
- Improved discard handling when the object map feature is enabled.
- rbd CLI ```import``` and ```copy``` commands now detect sparse and preserve sparse regions.
- Snapshots will now include a creation timestamp

**Comment by theanalyst** [comment]
@dillaman thanks! will update

**Comment by mattbenjamin** [comment]
@theanalyst hi abishek, for RGW, @pritha-srivastava noticed that we should have a note on the introduction of AWS bucket policy api support;  there's initial doc in doc/radosgw/bucketpolicy.rst; for now, policy is a means to express a range of new authorization concepts, in future, it is a building block for additional auth capabilities such as STS and group policy.

File: doc/release-notes.rst [hunk 1]
```diff
@@ -2,6 +2,112 @@
...
+    the management of encryption keys, automatic encryption (only recommended for
+    test setups), customer provided keys similar to Amazon SSE-C specification and
+    using a key management service (Openstack Barbician) similar to Amazon SSE-KMS
+    specification.
+  * RGW's metadata search with ElasticSearch now supports end user requests
+    serviced via RGW itself and now supports custom metadata fields
+  * RGW has consolidated the several metadata index pools via the use of rados
+    namespaces
+  * RGW now supports dynamic bucket index sharding
+
```
Comment (adamemerson) [inline doc/release-notes.rst]: Please add something like:

* RGW now has preliminary support for S3 style bucket policies.

**Pull request merged**